File: solutions/2026-04-22/java/2026-04-22-words-within-two-edits-of-dictionary.java
```java
import java.util.ArrayList;
import java.util.List;

// Problem: Words Within Two Edits of Dictionary
// Link: https://leetcode.com/problems/words-within-two-edits-of-dictionary/
//
// Approach:
// We need to iterate through each query word and compare it with every word in the dictionary.
// For each query word, we count the number of differing characters (edits) required to match a dictionary word.
// If the edit count is less than or equal to 2, we add the query word to our result list.
//
// Time Complexity:
// Let Q be the number of words in queries, D be the number of words in dictionary, and N be the length of each word.
// For each query word, we iterate through all D dictionary words.
// For each pair of query and dictionary words, we compare them character by character, which takes O(N) time.
// Therefore, the total time complexity is O(Q * D * N).
// Given the constraints (Q, D, N <= 100), the maximum operations would be around 100 * 100 * 100 = 1,000,000, which is efficient enough.
//
// Space Complexity:
// We use a list to store the result, which can store up to Q words.
// Therefore, the space complexity is O(Q) in the worst case.
class Solution {
    public List<String> twoEditWords(String[] queries, String[] dictionary) {
        List<String> result = new ArrayList<>();

        // Iterate through each query word
        for (String queryWord : queries) {
            boolean foundMatch = false; // Flag to check if the current query word matches any dictionary word within 2 edits

            // Iterate through each dictionary word to compare
            for (String dictWord : dictionary) {
                int edits = 0; // Counter for the number of edits needed

                // Compare the query word with the dictionary word character by character
                for (int i = 0; i < queryWord.length(); i++) {
                    // If characters at the same position are different, increment the edit count
                    if (queryWord.charAt(i) != dictWord.charAt(i)) {
                        edits++;
                    }
                    // Optimization: if edits exceed 2, we can stop comparing this pair
                    if (edits > 2) {
                        break;
                    }
                }

                // If the number of edits is 2 or less, we have found a match
                if (edits <= 2) {
                    result.add(queryWord); // Add the query word to the result list
                    foundMatch = true;     // Set the flag to true
                    break;                 // No need to check other dictionary words for this query, move to the next query
                }
            }
        }

        return result; // Return the list of query words that match
    }
}
```

File: solutions/2026-04-22/javascript/2026-04-22-words-within-two-edits-of-dictionary.js
// Problem: Words Within Two Edits of Dictionary
// Problem Summary: Find words from `queries` that can be transformed into words in `dictionary` with at most two character edits.
// Link: https://leetcode.com/problems/words-within-two-edits-of-dictionary/
// Approach Explanation:
// The problem requires us to check each query word against every dictionary word to see if they are within two edits (differences) of each other.
// Since the constraints are relatively small (up to 100 words, each up to 100 characters), a brute-force approach will be efficient enough.
// For each query word, we iterate through all dictionary words. For each pair, we calculate the number of differing characters (Hamming distance for strings of equal length).
// If the number of differences is less than or equal to 2, we consider the query word a match and add it to our result list.
//
// Time Complexity Analysis:
// Let Q be the number of words in `queries`, D be the number of words in `dictionary`, and N be the length of each word.
// For each of the Q query words, we iterate through all D dictionary words.
// For each pair of words, we compare them character by character, which takes O(N) time.
// Therefore, the total time complexity is O(Q * D * N).
// Given the constraints (Q, D <= 100, N <= 100), the maximum operations would be around 100 * 100 * 100 = 1,000,000, which is well within typical time limits.
//
// Space Complexity Analysis:
// The space complexity is dominated by the storage required for the result list. In the worst case, all query words might match.
// Therefore, the space complexity is O(Q), where Q is the number of query words.
// If we consider the input storage, it would be O(Q*N + D*N), but we usually refer to auxiliary space complexity.

/**
 * @param {string[]} queries
 * @param {string[]} dictionary
 * @return {string[]}
 */
var twoEditWords = function(queries, dictionary) {
    // This array will store the queries that match the criteria.
    const result = [];

    // Iterate through each word in the queries array.
    for (const queryWord of queries) {
        // Flag to check if the current queryWord has found a match in the dictionary.
        let foundMatch = false;

        // Iterate through each word in the dictionary array.
        for (const dictWord of dictionary) {
            // Calculate the number of differing characters between queryWord and dictWord.
            let diffCount = 0;
            // All words have the same length, so we can iterate up to that length.
            for (let i = 0; i < queryWord.length; i++) {
                // If characters at the same position are different, increment the difference count.
                if (queryWord[i] !== dictWord[i]) {
                    diffCount++;
                }
            }

            // If the number of differences is 2 or less, this queryWord is a match.
            if (diffCount <= 2) {
                // Mark that a match has been found for this queryWord.
                foundMatch = true;
                // Since we only need to know *if* it matches *any* dictionary word,
                // we can break out of the inner loop once a match is found.
                break;
            }
        }

        // If a match was found for the current queryWord, add it to the result list.
        if (foundMatch) {
            result.push(queryWord);
        }
    }

    // Return the list of query words that matched with at most two edits.
    return result;
};
```